### PR TITLE
GiveOfferController/ Logging/ Issue 101

### PR DESCRIPTION
**Pull request by anish97IND**

### CODE REVIEW

**Comment by magwas** [comment]
Updated your branch. Please pull before push.

File: src/main/java/com/kodekonveyor/work_request/offer/GiveofferController.java [hunk 1]
```diff
@@ -20,17 +22,23 @@
 
   @Autowired
   AuthenticatedUserService authenticatedUserService;
+  @Autowired
+  Logger loggerService; //NOPMD
```
Comment (magwas) [inline src/main/java/com/kodekonveyor/work_request/offer/GiveofferController.java:26]: Why this NOPMD?

File: src/test/java/com/kodekonveyor/work_request/offer/GiveOfferControllerLoggingTest.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    extends GiveofferControllerTestBase {
+
+  @BeforeEach
+  public void setUpTest() {
+    giveofferController.call(OfferDTOTestData.get());
+  }
+
+  @Test
+  @DisplayName(
+    "The call of GiveOfferController is Logged"
```
Comment (magwas) [inline src/test/java/com/kodekonveyor/work_request/offer/GiveOfferControllerLoggingTest.java:35]: You want to earn much more with this task.
There are lot more tests in this task. Please turn all relevant parts of the task description to tests.

Comment (anish97IND) [reply src/test/java/com/kodekonveyor/work_request/offer/GiveOfferControllerLoggingTest.java:35]: > You want to earn much more with this task.
> There are lot more tests in this task. Please turn all relevant parts of the task description to tests.

more test cases will be applied..

File: src/main/java/com/kodekonveyor/work_request/offer/GiveofferController.java (hunk 1 above)
Comment (anish97IND) [reply src/main/java/com/kodekonveyor/work_request/offer/GiveofferController.java:26]: its in reference with the issue faced in RevokeWorkRequestLogging (pending second review)...tried resolving it...howver, violation doesn't flag up locally.

Comment (magwas) [reply src/main/java/com/kodekonveyor/work_request/offer/GiveofferController.java:26]: please remove this and commit.

File: src/main/java/com/kodekonveyor/work_request/offer/GiveofferController.java [hunk 3]
```diff
@@ -20,11 +22,21 @@
...
 
   @PostMapping("/offer")
   public OfferDTO
       call(@RequestBody final OfferDTO offerDTO) {
+    loggerService.info(
+        WorkRequestConstants.SERVICE_CALL_NAME, this.getClass().getName()
+    );
+    loggerService.info(WorkRequestConstants.INPUT_VALIDATION, offerDTO.getId());
     OfferValidationUtil.inputValidation(workRequestRepository, offerDTO);
+    loggerService.debug(
```
Comment (magwas) [inline src/main/java/com/kodekonveyor/work_request/offer/GiveofferController.java:36]: no need to log each step. Only the start, the return, and the throws.

Comment (anish97IND) [reply src/main/java/com/kodekonveyor/work_request/offer/GiveofferController.java:36]: In this class all the validation are done in different class (InputValidationUtil.java). Do I need to log those?

**Comment by anish97IND** [comment]
> Good work so far, but some things to fix.

I am waiting for PMD issue to be resolved in shippable.

File: src/main/java/com/kodekonveyor/work_request/offer/GiveofferController.java (hunk 3 above)
Comment (magwas) [reply src/main/java/com/kodekonveyor/work_request/offer/GiveofferController.java:36]: No, for functionality we think in services and controllers. The code layout is implementation detail.

**Comment by magwas** [comment]
I see no issue in shippable.

**Comment by magwas** [comment]
Please remove //NOPMD comments to see if there is an issue with them is shippable.

**Comment by magwas** [comment]
updated, please pull before push

**Comment by szirbucz** [comment]
@magwas, please check your comments if they are still actual.

**Comment by magwas** [comment]
updated, please pull before push

**Comment by magwas** [comment]
testcases=6
please fill in https://docs.google.com/forms/d/e/1FAIpQLSdNMbUDPHp2CCbsKtx3qMDCGXIYmmRsjJoF201Xtl-xvb2llQ/viewform

**Pull request merged**